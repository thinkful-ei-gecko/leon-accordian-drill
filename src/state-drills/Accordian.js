import React, {Component} from 'react';

class Accordian extends React.Component {

  state = {
    currentClicked : null
  }

  accordianFeature(index) {
    this.setState({
      currentClicked : index
    })
  }

  renderLis() {
    return this.props.sections.map((section, index) => (
      <li key={index}>
        <button type="button" onClick={() => this.accordianFeature(index)}>{section.title}</button>
        <p>{this.state.currentClicked === index && section.content}</p>
      </li>
    ));
  }

  render() {

    return(
      <div>
        <ul>
          {this.renderLis()}
        </ul>

      </div>
    );
  }

}

export default Accordian;