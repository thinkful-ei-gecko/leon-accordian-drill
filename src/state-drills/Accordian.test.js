import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordian from './Accordian'

describe('<Accordian />', () => {


  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];

  //smoke test
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  //snapshots
  it('renders an empty ul with no section prop', () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordian sections={sections}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('clicking a button opens that paragraph', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only the last clicked section displays', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })







    
  // it('renders without errors', () => {
  //   const div = document.createElement('div');
  //   ReactDOM.render(<Accordian />, div);
  //   ReactDOM.unmountComponentAtNode(div);
  // })

  // it('renders an empty li when the sections prop is not supplied', () => {
  //   const wrapper = shallow(<Accordian />) 
  //   toJson(wrapper).toMatchSnapshot()
  // })

})