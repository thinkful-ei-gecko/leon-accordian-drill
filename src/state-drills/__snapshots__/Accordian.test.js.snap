// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordian /> clicking a button opens that paragraph 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`<Accordian /> only the last clicked section displays 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
      <p />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Accordian /> renders an empty ul with no section prop 1`] = `
<div>
  <ul />
</div>
`;

exports[`<Accordian /> renders no sections as active by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
      <p />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;
